src/plugins/rpd/rpdplugin.cpp

```diff
@@ -0,0 +1,319 @@
+                ++it;
+            }
+
+            mapJson.insert("decoName", decoName);
+            mapJson.insert("decoDesc", decoDesc);
+        }
+
+        if (layer->name() == "objects") {
+            QMap<QString, QJsonArray> objects;
+
+            for (auto object : layer->asObjectGroup()->objects()) {
+                QJsonObject desc;
+
+                desc.insert("kind", object->name());
+
+                desc.insert("x", qFloor(object->x() / 16.));
+                desc.insert("y", qFloor(object->y() / 16.));
+
+                auto properties = object->properties();
+                for (auto i = properties.begin(); i != properties.end(); ++i) {
+                    auto jsonCandidate = QJsonDocument::fromJson(i.value().toString().toUtf8());
+
+                    if (jsonCandidate.isObject()) {
+                        desc.insert(i.key(), jsonCandidate.object());
+                        continue;
+                    }
+
+                    desc.insert(i.key(), i.value().toJsonValue());
+                }
+
+                objects[object->type()].append(desc);
+            }
+
+            for (auto key : objects.keys())
+                mapJson.insert(key + "s", objects[key]);
+        }
+    }
+
+    if (!mapJson.contains("tiles"))
+        mapJson.insert("tiles", "tiles0_x.png");
+
+    mapJson.insert("water", "water0.png");
+
+    JsonWriter writer;
+    writer.setAutoFormatting(!options.testFlag(WriteMinimized));
+
+    if (!writer.stringify(QJsonDocument(mapJson).toVariant())) {
+        // This can only happen due to coding error
+        mError = writer.errorString();
+        return false;
+    }
+
+    QTextStream out(file.device());
+    out << writer.result();
+
+    out.flush();
+
+    if (file.error() != QFileDevice::NoError) {
+        mError = tr("Error while writing file:\n%1").arg(file.errorString());
+        return false;
+    }
+
+    if (!file.commit()) {
+        mError = file.errorString();
+        return false;
+    }
+
+    return true;
+}
+
+QString RpdMapFormat::nameFilter() const
+{
+    return tr("Remixed Pixel Dungeon levels (*.json)");
+}
+
+QString RpdMapFormat::errorString() const
+{
+    return mError;
+}
+
+RpdTilesetFormat::RpdTilesetFormat(QObject *parent)
+    : Tiled::TilesetFormat(parent)
+{}
+
+Tiled::SharedTileset RpdTilesetFormat::read(const QString &fileName)
+{
+    Q_UNUSED(fileName)
+    return Tiled::SharedTileset();
+}
+
+bool RpdTilesetFormat::supportsFile(const QString &fileName) const
+{
+    Q_UNUSED(fileName)
+    return false;
+}
+
+bool RpdTilesetFormat::write(const Tiled::Tileset &tileset, const QString &fileName, Options options)
+{
+    Tiled::SaveFile file(fileName);
+
+    if (!file.open(QIODevice::WriteOnly | QIODevice::Text)) {
+        mError = tr("Could not open file for writing.");
+        return false;
+    }
+
+    Tiled::MapToVariantConverter converter;
+    QVariant variant = converter.toVariant(tileset, QFileInfo(fileName).dir());
```

Comment:
This looks like just the same thing as `JsonTilesetFormat`. Why do you need it again here?

Reply:
Well, I just copied and pasted it from JSON plugin template a long time ago (in another fork) and not touched it since. It not need and removed now.